be/src/olap/rowset/segment_v2/segment.cpp

```diff
@@ -259,9 +263,8 @@ Status Segment::new_column_iterator(const TabletColumn& tablet_column, ColumnIte
                         tablet_column.has_default_value(), tablet_column.default_value(),
                         tablet_column.is_nullable(), std::move(type_info), tablet_column.length(),
                         tablet_column.precision(), tablet_column.frac()));
-        ColumnIteratorOptions iter_opts;
 
-        RETURN_IF_ERROR(default_value_iter->init(iter_opts));
+        RETURN_IF_ERROR(default_value_iter->init(opts));
         *iter = default_value_iter.release();
         return Status::OK();
     }
```

Comment:
not only default iterator should call init here, also other column iterators on line 271 should call init here.